.github/workflows/build.yml

```diff
@@ -0,0 +1,113 @@
+name: Scala tests with coverage
+
+on:
+  pull_request:
+    paths-ignore: ['**.md']
+  push:
+    paths-ignore: ['**.md']
```

Comment:
not sure why, but the tests for this PR seem to be running duplicate tests for each(?) commit, one for push and one for pull_request? maybe we really just need `push` if that runs on all pushed commits. not a big deal if we like it this way

```
Scala tests with coverage / test (pull_request) Successful in 6m
Scala tests with coverage / test (push) Successful in 7m

Scala tests with coverage / generateAndPublishClient (pull_request) Successful in 1m
Scala tests with coverage / generateAndPublishClient (push) Successful in 1m
```

Reply:
IDK, but it mirrors what travis did (with two identical test runs), and I am trying to keep it as similar as possible for the moment. I think we can make later improvements down the line if it makes sense to.